File: nomenclator/src/main/java/flowmanager/nomenclator/model/Comment.java
package flowmanager.nomenclator.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(exclude = {
        "author",
        "workItem"
})
@Builder
@Entity
@Table(name = "comment")
public class Comment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable = false)
    private String content;

    @Column(nullable = false)
    private LocalDateTime createdAt;

    @Column
    private LocalDateTime updatedAt;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User author;

    @ManyToOne
    @JoinColumn(name = "work_item_id", nullable = false)
    private WorkItem workItem;
}

File: frontend/app/types/comment.ts
import type { UserSummaryDto } from "./user"
import type { WorkItemSummaryDto } from "./workItem"

export interface CommentCreateDto {
  content: string
  workItemId: number
}

export interface CommentUpdateDto {
  content: string
}

export interface CommentResponseDto {
  id: number
  content: string
  createdAt: string
  updatedAt: string
  author: UserSummaryDto
  workItem: WorkItemSummaryDto
}

export interface CommentResponseWorkItemDto {
  id: number
  content: string
  createdAt: string
  updatedAt: string
  author: UserSummaryDto
}